tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -1617,3 +1623,102 @@ def test_server_pod_restart_kubectl_delete_retries(self):
+        LOGGER.info("Step 1: Successfully got the endpoint to be used for IOs: %s", endpoint)
+
+        LOGGER.info("Step 2: Start IOs with variable object sizes in background")
+        test_prefix = f'test-44855-{t_t}'
+        output = Queue()
+        event_set_clr = [False]
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': test_prefix,
+                'nclients': 2, 'nsamples': 10, 'output': output, 'event_set_clr': event_set_clr,
+                'connect_timeout': HA_CFG["common_params"]["1sec_delay"],
+                'end_point': endpoint, 'host': self.master_host, "user": self.master_user,
+                "pwd": self.master_pwd}
+        thread = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 2: Successfully started IOs with variable sizes objects in background")
+        LOGGER.info("Waiting for %s seconds", HA_CFG["common_params"]["30sec_delay"])
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
+
+        LOGGER.info("Step 3: Restart the server pod by kubectl delete.")
+        LOGGER.info("Deleting pod %s", pod_name)
+        event.set()
+        resp = self.node_master_list[0].delete_pod(pod_name=pod_name, force=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to delete pod {pod_name} by kubectl delete")
+        LOGGER.info("Step 3: Successfully restarted pod %s by kubectl delete", pod_name)
+
+        LOGGER.info("Step 4: Check cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0])
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 4: Cluster is in good state")
+
+        LOGGER.info("Waiting for thread to join")
+        thread.join()
+        event.clear()
+
+        LOGGER.info("Step 5: Verify status for In-flight IOs")
+        responses = dict()
+        while len(responses) != 2:
+            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in responses["pass_res"])
+        fail_logs = list(x[1] for x in responses["fail_res"])
+        all_logs = pass_logs + fail_logs
+        LOGGER.info("Copy s3bench log files to local system")
+        for file in all_logs:
+            resp = self.node_master_list[0].copy_file_to_local(remote_path=file, local_path=file)
+            assert_utils.assert_true(resp[0], resp[1])
+        resp = self.ha_obj.check_s3bench_log(file_paths=pass_logs)
+        assert_utils.assert_false(len(resp[1]), "IOs before pod deletion are expected to pass. "
+                                                f"Logs which contain failures: {resp[1]}")
+        resp = self.ha_obj.check_s3bench_log(file_paths=fail_logs, pass_logs=False)
+        assert_utils.assert_false(len(resp[1]), f"IOs after pod deletion are expected to "
```

Comment:
weren't u saying sometimes it doesn't fail immediately?

Reply:
We have added connect_timeout for that.